src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -416,26 +388,25 @@ public void Dispose()
         {
             this.logger.LogTrace("()");
 
-            if (!this.serverCancel.IsCancellationRequested)
-            {
-                this.serverCancel.Cancel();
+            this.serverCancel.Cancel();
 
-                this.logger.LogTrace("Stopping network peer server.");
-                this.listener.Dispose();
+            this.logger.LogTrace("Stopping network peer server.");
+            this.listener.Dispose();
 
-                try
-                {
-                    this.ConnectedNetworkPeers.DisconnectAll();
-                }
-                finally
-                {
-                    if (this.socket != null)
-                    {
-                        Utils.SafeCloseSocket(this.socket);
-                        this.socket = null;
-                    }
-                }
+            try
+            {
+                this.ConnectedNetworkPeers.DisconnectAll();
             }
+            catch (Exception e)
+            {
+                this.logger.LogTrace("Exception occurred: {0}", e.ToString());
+            }
+
+            this.tcpListener?.Stop();
+
+            this.logger.LogTrace("Waiting for {0} unfinished tasks to complete.", this.unfinishedTasks);
```

Comment:
Did you mean to show the number of tasks we're waiting on?

Reply:
yes, that was just debug thing for me, as mentioned, pls ignore unfinished tasks, they are removed i think in part 11